crates/bevy_input/src/gamepad.rs

```diff
@@ -473,114 +494,385 @@ impl GamepadSettings {
+    pub fn try_set_press_threshold(&mut self, value: f32) -> Result<()> {
+        if (self.release_threshold..=1.0).contains(&value) {
+            self.press_threshold = value;
+            Ok(())
+        } else if !(0.0..1.0).contains(&value) {
+            Err(GamepadSettingsError::InvalidButtonSettingPressThresholdOutOfRange(value))
+        } else {
+            Err(GamepadSettingsError::InvalidButtonSettingReleaseThresholdGreaterThanPressThreshold { press_threshold: value, release_threshold: self.release_threshold })
+        }
+    }
+
+    /// Try to set the button input threshold above which the button is considered pressed.
+    /// If the value passed is outside the range [`release_threshold`, 1.0], the value will not be changed.
+    ///
+    /// Returns the new value of `press_threshold`.
+    pub fn set_press_threshold(&mut self, value: f32) -> f32 {
+        self.try_set_press_threshold(value).ok();
+        self.press_threshold
+    }
+
+    /// Get the button input threshold below which the button is considered released.
+    pub fn release_threshold(&self) -> f32 {
+        self.release_threshold
+    }
+
+    /// Try to set the button input threshold below which the button is considered released.
+    ///
+    /// # Errors
+    ///
+    /// If the value passed is outside the range [0.0, `press_threshold`], returns
+    /// `GamepadSettingsError::InvalidButtonSettingReleaseThresholdOutOfRange` or
+    /// `GamepadSettingsError::InvalidButtonSettingReleaseThresholdGreaterThanPressThreshold`.
+    pub fn try_set_release_threshold(&mut self, value: f32) -> Result<()> {
+        if (0.0..=self.press_threshold).contains(&value) {
+            self.release_threshold = value;
+            Ok(())
+        } else if !(0.0..1.0).contains(&value) {
+            Err(GamepadSettingsError::InvalidButtonSettingReleaseThresholdOutOfRange(value))
+        } else {
+            Err(GamepadSettingsError::InvalidButtonSettingReleaseThresholdGreaterThanPressThreshold { press_threshold: self.press_threshold, release_threshold: value })
+        }
+    }
+
+    /// Try to set the button input threshold below which the button is considered released. If the
+    /// value passed is outside the range [0.0, `press_threshold`], the value will not be changed.
+    ///
+    /// Returns the new value of `release_threshold`.
+    pub fn set_release_threshold(&mut self, value: f32) -> f32 {
+        self.try_set_release_threshold(value).ok();
+        self.release_threshold
     }
 }
 
 /// Settings for a [`GamepadAxis`].
 ///
 /// It is used inside of the [`GamepadSettings`] to define the sensitivity range and
 /// threshold for an axis.
+/// Values that are higher than `livezone_upperbound` will be rounded up to 1.0.
+/// Values that are lower than `livezone_lowerbound` will be rounded down to -1.0.
+/// Values that are in-between `deadzone_lowerbound` and `deadzone_upperbound` will be rounded
```

Comment:
those fields are now private, putting their exact name instead of a sentence is more confusing

Reply:
Stopped quoting them.